bindings/rust/wasmedge-sys/src/instance/table.rs

```diff
@@ -153,10 +154,16 @@ impl Table {
 }
 impl Drop for Table {
     fn drop(&mut self) {
-        if !self.registered && !self.inner.0.is_null() {
-            unsafe {
-                ffi::WasmEdge_TableInstanceDelete(self.inner.0);
-            }
+        if !self.registered && Arc::strong_count(&self.inner) == 1 && !self.inner.0.is_null() {
+            unsafe { ffi::WasmEdge_TableInstanceDelete(self.inner.0) };
+        }
+    }
+}
+impl Clone for Table {
+    fn clone(&self) -> Self {
+        Table {
+            inner: self.inner.clone(),
```

Comment:
It looks like the `clone` will only copy the pointer. Is this a expected behavior?

Reply:
Actually, the `clone` operation does not copy the pointer. The type of `self.inner` is `Arc<InnerTable>`, and `Arc<T>` is a reference type and uses an inner reference counter to record how many references it has at some time. The `clone` method of `Arc<T>` just increments the counter, but does not copy the low-level raw pointer. Thanks!

Reply:
Good to know. Thanks.